mlflow/azure/client.py

```diff
@@ -0,0 +1,152 @@
+"""
+This module provides utilities for performing Azure Blob Storage operations without requiring
+the heavyweight azure-storage-blob library dependency
+"""
+from copy import deepcopy
+import urllib
+import logging
+
+from mlflow.utils import rest_utils
+
+_logger = logging.getLogger(__name__)
+_PUT_BLOCK_HEADERS = {
+    "x-ms-blob-type": "BlockBlob",
+}
+
+
+def put_block(sas_url, block_id, data, headers):
+    """
+    Performs an Azure `Put Block` operation
+    (https://docs.microsoft.com/en-us/rest/api/storageservices/put-block)
+
+    :param sas_url: A shared access signature URL referring to the Azure Block Blob
+                    to which the specified data should be staged.
+    :param block_id: A base64-encoded string identifying the block.
+    :param data: Data to include in the Put Block request body.
+    :param headers: Additional headers to include in the Put Block request body
+                    (the `x-ms-blob-type` header is always included automatically).
+    """
+    request_url = _append_query_parameters(sas_url, {"comp": "block", "blockid": block_id})
+
+    request_headers = deepcopy(_PUT_BLOCK_HEADERS)
+    for name, value in headers.items():
+        if _is_valid_put_block_header(name):
```

Comment:
I've introduced some client-side header name validation for the `put_block_list` and `put_block` operations after encountering an issue introduced by a recent Databricks MLflow change where the `x-ms-blob-type` header is returned as part of the SAS URI credentials from the `GetCredentialsForWrite` operation; the current client logic in this PR passes this `x-ms-blob-type` header to `put_block_list` operation (https://github.com/mlflow/mlflow/blob/125747259f3d85ad83ac838db87f1d2b5c715659/mlflow/store/artifact/databricks_artifact_repo.py#L188), but `x-ms-blob-type` isn't supported for Put Block List, which causes the request to fail. As a result, uploads on Azure Databricks throw exceptions when this client is run with the latest MLflow service code.

@WeichenXu123 can you add some unit tests for `mlflow/azure/client.py` to make sure that unsupported headers are removed prior to invocation of `cloud_storage_http_request`?